routing_model/src/state.rs

```diff
@@ -14,10 +14,16 @@ use crate::flows_node::*;
 use crate::flows_src::*;
 use crate::utilities::*;
 
+#[derive(Debug, PartialEq, Clone)]
+pub struct ProcessElderChangeState {
+    pub change_elder: ChangeElder,
+}
+
 #[derive(Debug, PartialEq, Default, Clone)]
 pub struct CheckAndProcessElderChangeState {
     pub wait_votes: Vec<ParsecVote>,
     pub change_elder: Option<ChangeElder>,
+    pub sub_routine_process_elder_change: Option<ProcessElderChangeState>,
```

Comment:
The alternative to replace the boiler plate sub routine would be:

```
#[derive(Debug, PartialEq, Clone)]
pub struct ProcessElderChangeState {
    is_active: bool,
    pub change_elder: ChangeElder,
}

#[derive(Debug, PartialEq, Default, Clone)]
pub struct CheckAndProcessElderChangeState {
    pub wait_votes: Vec<ParsecVote>,
    pub sub_routine_process_elder_change: ProcessElderChangeState,
}
```

```
impl ProcessElderChange {
    pub fn start_event_loop(&self, change_elder: ChangeElder) -> Self {
        let state = self.clone();
        state.mut_routine_state().is_active = true;
        state.mut_routine_state().change_elder Some(change_elder);
        state.vote_for_elder_change()
    }

    fn routine_state(&self) -> &ProcessElderChangeState {
        &self.0.check_and_process_elder_change_routine.sub_routine_process_elder_change
    }

    fn mut_routine_state(&mut self) -> &mut ProcessElderChangeState {
        &mut self.0.check_and_process_elder_change_routine.sub_routine_process_elder_change
    }

```

Reply:
so wait_votes, I see you kept it in `CheckAndProcessElderChangeState` yet it only appears in `ProcessElderChange`. Any reason for this?

Reply:
got it wrong. fixed in the comment.

Reply:
But yeah, I think we should simplify to something like this at some point